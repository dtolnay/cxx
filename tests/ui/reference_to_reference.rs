#[cxx::bridge]
mod ffi {
    unsafe extern "C++" {
        type ThingC;
        fn ref_c(t: &ThingC);
        fn rvalue_c(t: &&ThingC);
        fn repro_c(t: &&&ThingC);
    }
    extern "Rust" {
        type ThingR;
        fn ref_r(t: &ThingC);
        fn rvalue_r(t: &ThingC);
        fn repro_r(t: &&&ThingR);
    }
}

fn main() {}
