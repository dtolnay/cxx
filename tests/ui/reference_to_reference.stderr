error: C++ does not allow references to references
 --> $DIR/reference_to_reference.rs:7:23
  |
7 |         fn repro_c(t: &&&ThingC);
  |                       ^^^^^^^^^

error: C++ does not allow references to references
  --> $DIR/reference_to_reference.rs:13:23
   |
13 |         fn repro_r(t: &&&ThingR);
   |                       ^^^^^^^^^
